### PR TITLE
WIP Extract key from new StorageClass SecretRefs attribue

### PR DESCRIPTION
**Pull request by sbezverk**
PR adds code to retrieve secret name and namespace from new proposed first class attribute in StorageClass. This will allow external-provisioner stays backend independent from the perspective of processing backend specific credentials.

### CODE REVIEW

File: pkg/controller/controller.go [hunk 1]
```diff
@@ -203,6 +232,28 @@ func (p *csiProvisioner) Provision(options controller.VolumeOptions) (*v1.Persis
 	capacity := options.PVC.Spec.Resources.Requests[v1.ResourceName(v1.ResourceStorage)]
 	volSizeBytes := capacity.Value()
 
+	userCredentials := map[string]string{}
+	storageClassName := options.PVC.Spec.StorageClassName
+	fmt.Printf("><SB> provisioner was called for storage class: %s\n", *storageClassName)
+	storageClass, err := p.client.StorageV1().StorageClasses().Get(*storageClassName, metav1.GetOptions{})
+	if err != nil {
+		glog.Warningf("failed to retrieve information about the storage class: %s with error: %v", storageClassName, err)
+	} else {
```
Comment (vladimirvivien) [inline pkg/controller/controller.go]: Uneccessary else-block.

File: pkg/controller/controller.go [hunk 2]
```diff
@@ -203,6 +232,28 @@ func (p *csiProvisioner) Provision(options controller.VolumeOptions) (*v1.Persis
 	capacity := options.PVC.Spec.Resources.Requests[v1.ResourceName(v1.ResourceStorage)]
 	volSizeBytes := capacity.Value()
 
+	userCredentials := map[string]string{}
+	storageClassName := options.PVC.Spec.StorageClassName
+	fmt.Printf("><SB> provisioner was called for storage class: %s\n", *storageClassName)
```
Comment (vladimirvivien) [inline pkg/controller/controller.go]: Remember to remove fmt.Printf statements

Comment (sbezverk) [reply pkg/controller/controller.go]: Thanks, will do.

File: pkg/controller/controller.go (hunk 1 above)
Comment (sbezverk) [reply pkg/controller/controller.go]: Appreciate if you could clarify why. The logic here is if we cannot get a storage class object, we just print warning and do not process any secret info, basically just skipping whatever is covered by else part.

Comment (vladimirvivien) [reply pkg/controller/controller.go]: It looks like storageclass is optional.  If so, then the code is good.

File: pkg/controller/controller.go [hunk 3]
```diff
@@ -203,6 +232,28 @@ func (p *csiProvisioner) Provision(options controller.VolumeOptions) (*v1.Persis
...
+	storageClass, err := p.client.StorageV1().StorageClasses().Get(*storageClassName, metav1.GetOptions{})
```
Comment (rootfs) [inline pkg/controller/controller.go]: validate storageClassName

**Comment by sbezverk** [comment]
implemented via another PR

**Pull request closed without merging**